src/main/scala/com/codergaden/ex1/Application.scala

```diff
@@ -0,0 +1,55 @@
+package com.codergaden.ex1
+
+object Application {
+
+  def main(args: Array[String]): Unit = {
+    val res = getShorterLonger(args.toList)
+    println(
+      s"""Shorter word/s: ${res.shortestWords.words.mkString(" ")} with num letters ${res.shortestWords.lengthOfWord}
+      |Longer word/s: ${res.longestWords.words.mkString(" ")} with num letters ${res.longestWords.lengthOfWord}""".stripMargin )
+  }
+
+  def getShorterLonger(wordlist: List[String]): (Result) = {
+    wordlist.foldLeft (Result.empty) {
+      case (res@Result(shortestWords, longestWords), currWord) =>
+        currWord match {
+          case word if word.length < WordsWithLength.minLength => res
+          case other => Result(ShortestWords(shortestWords).appendOrPut(other), LongestWords(longestWords).appendOrPut(other))
+        }
+    }
+  }
+}
+
+case class ShortestWords(words: WordsWithLength) {

```

Comment:
instead of wrapping WordsWithLength you can have a
sealed trait Words{ def words: List[String] def appendOrPut: Words def lengthOfWord: Int = words.headOption.map(_.length).getOrElse(0) }

and then having
case class ShortestWords(words: List[String]) extends Words { ...implementation here }

and same thing for Longest words so you can improve the data structure and remove one wrap